### PR TITLE
Integrated new uprotocol-java SDK in UDiscovery

### PR DESCRIPTION
**Pull request by jvnaiduwin**
Refactored unit tests
Remove synchronization with Central Discovery Service

-factory reset initialization
-intersection logic
-remote RPC calls
-Remove uRegistry Database, hardcoded URI strings, Authenticator class, Tag class, Key class, Log class, ultifi key name and unwanted permission. -UDiscovery Unit Test update
-Refactor variable names
-Android entry for entity name and version
-Unit Test coverage in total approx 100%
-Change artifactory name org.eclipse.uprotocol.core.udiscovery. -UDiscovery code cleanup.
-Changed gradle dependency to reflect new names of repos. -Contribution doc updated.
-Class rename to ResourceLoader.
-Change invokeMethod() to return UMessage

### CODE REVIEW

**Review by stevenhartley — CHANGES_REQUESTED** [review]
@jvnaiduwin please update to up-java 1.5.8 and the latest from up-client-android-java as I cannot run this pr on the emulator now.

File: gradle/libs.versions.toml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+androidx-ktx = "1.6.0"
+junit = "4.13.2"
+kotlin = "1.9.21"
+kotlinx-coroutines = "1.7.1"
+ksp = "1.9.21-1.0.15"
+mockito = "4.6.1"
+robolectric = "4.7.3"
+room = "2.4.2"
+room-testing = "1.1.1"
+up-java = "1.5.7-SNAPSHOT"
```
Comment (stevenhartley) [inline gradle/libs.versions.toml]: change to 1.5.8

File: service/src/main/java/org/eclipse/uprotocol/core/udiscovery/UDiscoveryService.java [hunk 2]
```diff
@@ -0,0 +1,462 @@
...
+import org.eclipse.uprotocol.UPClient;
+import org.eclipse.uprotocol.common.UStatusException;
+import org.eclipse.uprotocol.common.util.log.Key;
+import org.eclipse.uprotocol.core.udiscovery.db.DiscoveryManager;
+import org.eclipse.uprotocol.core.udiscovery.v3.FindNodePropertiesResponse;
+import org.eclipse.uprotocol.core.udiscovery.v3.FindNodesResponse;
+import org.eclipse.uprotocol.core.udiscovery.v3.LookupUriResponse;
+import org.eclipse.uprotocol.rpc.CallOptions;
+import org.eclipse.uprotocol.rpc.URpcListener;
+import org.eclipse.uprotocol.uri.builder.UResourceBuilder;
```
Comment (stevenhartley) [inline service/src/main/java/org/eclipse/uprotocol/core/udiscovery/UDiscoveryService.java]: the package was renamed to factory per request from Mikael

**Review by stevenhartley — CHANGES_REQUESTED** [review]

File: gradle/libs.versions.toml (hunk 1 above)
Comment (jvnaiduwin) [reply gradle/libs.versions.toml]: Done

File: service/src/main/java/org/eclipse/uprotocol/core/udiscovery/UDiscoveryService.java (hunk 2 above)
Comment (jvnaiduwin) [reply service/src/main/java/org/eclipse/uprotocol/core/udiscovery/UDiscoveryService.java]: Done

**Review by jvnaiduwin — COMMENTED** [review]
Changes done.

**Review by stevenhartley — APPROVED** [review]

**Review by kz9nyy — APPROVED** [review]

**Pull request merged**